SU-SWS/stanford_ssp
# SITES-993: infinite redirect; additive restrictions

See #70 and #94.

# READY FOR REVIEW 

# Summary
- Provides authorization checks by sunet or workgroup
- If the configuration in either field fails validation the users are denied. 

# Needed By (Date)
- Thursday

# Urgency
- High

# Steps to Test Authorization

1. Copy a site with no anonymous access to content to your local
2. Ensure that a workgroup value is set that your account is not in (/admin/config/stanford/stanford_ssp/authorizations)
3. Try to log in and see you are successful. 
4. Check out this branch
5. Log out and log back in again and see you are unsuccessful. 

# Steps to test 💩  loop

1. Open two windows with one incognito
2. Check out this branch on a recent build of the cardinald7 stack or a cloned site
2. Log in as User 1 on one of the windows
3. Set the configuration to restrict access to just your sunet at /admin/config/stanford/stanford_ssp/authorizations
4. Log in as your weblogin user on the other window
5. Log out from the weblogin user
6. Remove your sunet from the authorizations list and add some other sunetid 
7. Try to log in with your weblogin user again and see an access denied page (not a loop)
8. Remove all sunetid restrictions and add an arbitrary group to the workgroup restrictions
9. Try to log in with your weblogin user again and see an access denied page (not a loop)
10. Remove the arbitrary workgroup restriction and replace with `uit:sws`
11. Log in successfully with your weblogin user

# Affected Projects or Products
- Everything

# Associated Issues and/or People
- SITES-753
- #59 
- @rebeccahongsf 
- SITES-993
- #70 
- #94 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
